mypy/plugin.py

```diff
@@ -159,7 +162,10 @@ def qualified_name(self, n: str) -> str:
 MethodContext = NamedTuple(
     'MethodContext', [
         ('type', Type),                    # Base object type for method call
-        ('arg_types', List[List[Type]]),
+        ('callee_arg_names', List[Optional[str]]),
+        ('arg_names', List[List[Optional[str]]]),
+        ('arg_kinds', List[List[int]]),
+        ('arg_types', List[List[Type]]),  # List of actual caller types for each formal argument
         ('default_return_type', Type),
         ('args', List[List[Expression]]),
         ('context', Context),
```

Comment:
Could you please also add a comment for every existing attribute here and above?